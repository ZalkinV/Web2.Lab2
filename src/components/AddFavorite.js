import React from "react";
import { connect } from "react-redux";
import { addFavorite } from "../actions";


class AddFavorite extends React.Component {
  render() {
    return (
      <form onSubmit={(e) => this.handleSubmit(e)}>
        <input type="text" name="cityName" placeholder="City name" required={true} />
        <input type="submit" value="Add to favorite"/>
      </form>
    );
  }

  handleSubmit(e) {
    e.preventDefault();

    this.props.addFavorite(e.currentTarget.elements.cityName.value);
  }
}

function mapDispatchToProps(dispatch) {
  return {
    addFavorite: (cityName) => {
      dispatch(addFavorite(cityName));
    }
  };
}

export default connect(null, mapDispatchToProps)(AddFavorite);
