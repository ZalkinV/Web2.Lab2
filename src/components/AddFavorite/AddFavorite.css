.add-favorite {
  display: inline;
}

.add-favorite > .input {
  display: inline;
}
