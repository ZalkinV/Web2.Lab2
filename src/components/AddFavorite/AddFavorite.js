import React from "react";


export default function AddFavorite(props) {
  return (
    <form class="add-favorite" onSubmit={(e) => props.onSubmit(e)}>
      <input class="input" type="text" name="cityName" placeholder="City name" required={true} />
      <input class="button" type="submit" value="Add to favorite"/>
    </form>
  );
}
