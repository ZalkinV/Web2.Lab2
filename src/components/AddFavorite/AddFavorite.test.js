import React from "react";
import renderer from "react-test-renderer";

import AddFavorite from "./AddFavorite";


describe("<AddFavorite /> render", () => {

  test("should has form with text input and button", () => {
    const tree = renderer
      .create(<AddFavorite />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
