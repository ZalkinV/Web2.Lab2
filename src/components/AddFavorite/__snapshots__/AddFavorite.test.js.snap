// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddFavorite /> render should has form with text input and button 1`] = `
<form
  className="add-favorite"
  onSubmit={[Function]}
>
  <input
    className="input"
    name="cityName"
    placeholder="City name"
    required={true}
    type="text"
  />
  <input
    className="button"
    type="submit"
    value="Add to favorite"
  />
</form>
`;
